homeassistant/components/insteon/api/device.py

```diff
@@ -74,3 +83,70 @@ async def websocket_get_device(
         "aldb_status": str(device.aldb.status),
     }
     connection.send_result(msg[ID], device_info)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/device/add",
+        vol.Required(MULTIPLE): bool,
+        vol.Optional(DEVICE_ADDRESS): str,
+    }
+)
+@websocket_api.require_admin
+@websocket_api.async_response
+async def websocket_add_device(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict,
+) -> None:
+    """Add one or more Insteon devices."""
+    hass.async_create_task(
```

Comment:
Why create a task and not await it ?

Reply:
The nice thing about awaiting it is that in the frontend you know when the command is done.

Reply:
I am trying to give feedback to the user during the process so the UI does update based on backend events. I create the task so the adding of the device happens in the backend while the frontend listens for the events. This gives the user the ability to cancel the process too.